core/src/main/java/com/linecorp/armeria/server/misc/InetAddressPredicates.java

```diff
@@ -0,0 +1,199 @@
+            Pattern.compile("^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}" +
+                            "(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$");
+
+    /**
+     * A {@link Splitter} for splitting an IP address.
+     */
+    private static final Splitter dotSplitter = Splitter.on('.').limit(4);
+
+    /**
+     * A table for converting an IP decimal value to the number of bits.
+     */
+    private static final Map<Integer, Integer> ipDecimalToMaskBits = new Builder<Integer, Integer>()
+            .put(0xFF, 8)
+            .put(0xFE, 7)
+            .put(0xFC, 6)
+            .put(0xF8, 5)
+            .put(0xF0, 4)
+            .put(0xE0, 3)
+            .put(0xC0, 2)
+            .put(0x80, 1)
+            .put(0x00, 0)
+            .build();
+
+    /**
+     * Returns a {@link Predicate} which returns {@code true} if the given {@link InetAddress} equals to
+     * the specified {@code address}.
+     *
+     * @param address the expected {@link InetAddress}
+     */
+    public static Predicate<InetAddress> ofExact(InetAddress address) {
+        requireNonNull(address, "address");
+        return address::equals;
+    }
+
+    /**
+     * Returns a {@link Predicate} which returns {@code true} if the given {@link InetAddress} equals to
+     * the specified {@code address}.
+     *
+     * @param address the expected IP address string, e.g. {@code 10.0.0.1}
+     */
+    public static Predicate<InetAddress> ofExact(String address) {
+        requireNonNull(address, "address");
+        try {
+            final InetAddress inetAddress = InetAddress.getByName(address);
+            return inetAddress::equals;
+        } catch (UnknownHostException e) {
+            throw new IllegalArgumentException("Invalid address: " + address, e);
+        }
+    }
+
+    /**
+     * Returns a {@link Predicate} which returns {@code true} if the given {@link InetAddress} is in the
+     * range of a <a href="https://tools.ietf.org/html/rfc4632">Classless Inter-domain Routing (CIDR)</a> block.
+     *
+     * @param baseAddress the base {@link InetAddress} of a CIDR notation
+     * @param maskBits the number of significant bits which describes its network portion
+     */
+    public static Predicate<InetAddress> ofCidr(InetAddress baseAddress, int maskBits) {
+        requireNonNull(baseAddress, "baseAddress");
+        checkArgument(maskBits >= 0, "maskBits: %s (expected: >= 0)", maskBits);
+        return ofCidr(baseAddress, maskBits, maskBits);
+    }
+
+    /**
+     * Returns a {@link Predicate} which returns {@code true} if the given {@link InetAddress} is in the
+     * range of a <a href="https://tools.ietf.org/html/rfc4632">Classless Inter-domain Routing (CIDR)</a> block.
+     *
+     * @param baseAddress the base {@link InetAddress} of a CIDR notation
+     * @param subnetMask the subnet mask, e.g. {@code 255.255.255.0}
+     */
+    public static Predicate<InetAddress> ofCidr(InetAddress baseAddress, String subnetMask) {
```

Comment:
I added this method because [jboss.netty.handler.ipfilter.CIDR](https://github.com/netty/netty/blob/3.10/src/main/java/org/jboss/netty/handler/ipfilter/CIDR.java#L60) provides it. But I'm curious that is it necessary for us?
@trustin @minwoox Please let me know what you think! :-)

Reply:
I think it's good to have this one. There should be someone who prefers this representation to bits.

Reply:
:+1: